presto-docs/src/main/sphinx/admin/tuning.rst

```diff
@@ -10,29 +10,392 @@ Config Properties
++                                          +                                          +                                    +However in many cases increasing this value will       +
++                                          +                                          +                                    +visibly increase computation time while writing.       +
++------------------------------------------+------------------------------------------+------------------------------------+-------------------------------------------------------+
++        ``query.execution-policy``        +``String`` (``all-at-once`` or ``phased``)+          ``all-at-once``           +Serves as default value for ``execution_policy``       +
++                                          +                                          +                                    +session property. Setting this value to ``phased``     +
++                                          +                                          +                                    +will allow query scheduler to split a single query     +
++                                          +                                          +                                    +execution between different time slots. This will      +
++                                          +                                          +                                    +allow to switch context more often and possibly stage  +
++                                          +                                          +                                    +the partially executed query in order to increase      +
++                                          +                                          +                                    +robustness. Average time of executing query may        +
++                                          +                                          +                                    +slightly increase after setting this to ``phased`` due +
++                                          +                                          +                                    +to context switching and more complex scheduling       +
++                                          +                                          +                                    +algorithm but drop in variation of query execution     +
++                                          +                                          +                                    +time is expected.                                      +
++------------------------------------------+------------------------------------------+------------------------------------+-------------------------------------------------------+
++    ``query.initial-hash-partitions``     +               ``Integer``                +               ``8``                +Serves as default value for ``hash_partition_count``   +
```

Comment:
Once more I have the feeling that I'm not the right person for this task ;-)
Anyway, I was surprised to see hardcoded ``8`` here, but I just described what I found I don't feel sure enough to discuss with that.

I guess hardcoded value may be defended as safe value for not very large queries. Even if there are lots more of nodes in cluster if there are also many queries, then utilization of cluster should be fine.

My first intuition would be that it should be number of nodes in presto cluster by default, but it has some drawbacks too. So it's not an obvious choice for me.

I wrote that "number of presto nodes may lower the utilization of cluster" and "high value will cause assigning multiple partitions of same query to one node" and I think it covers how it works without discussing with the implementation to much.

I assume that in perfect world that should be adjusted based on workload, query size and more and as I understand partitioning systems other then ``FIXED`` are place for that.